adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,47 @@
+= Rename Reactive extensions
+
+* Status: proposed
+* Date: 2023-10-17
+
+== Context and Problem Statement
+
+RESTEasy Reactive is the preferred REST stack in Quarkus, however in the almost 3 years since its release, we have seen
+folks of all experience misunderstand its capabilities and being led to believe that it forces reactive programming on them.
+With the advent of Virtual Threads in Java 19+, the problem becomes larger since in the minds of users, Reactive and Virtual Threads
+are two different worlds.
+
+The same problem applies to other extensions as well:
+* Reactive Messaging
+* Reactive Routes
+* REST Client reactive
+
+The common theme here is that these extensions can work equally well with reactive and blocking code, but their name confuses users.
+
+We have tried to clarify the situation with blog posts, talks and on the various communication channels, but there seems to be
+no escaping the fact that "reactive" in the name primes people in a non-desirable way.
+
```

Comment:
We need to consider impact of breaking users here.

i.e. if the solution here break users existing applications/migration paths we hurt the goal of making updates easy and progressive rather than absolute breaking.

Reply:
The only thing potentially breaking here is the idea of renaming the rest client (see other comments)

Reply:
yes - but that should be in outlined/explained in the ADR. even from the comments I don't grok how big a breakage it would be.

Reply:
Once we figure out what to do, I can update the ADR